unit_tests/test_packages.py

```diff
@@ -20,6 +19,9 @@ def setUp(self):
         super(PackagesTest, self).setUp()
         self.apt = AptStub()
         self.packages = Packages(apt=self.apt)
+        keyfile = "jenkins.io.key"
+        os.symlink(os.path.join(os.getcwd(), keyfile),
+                   os.path.join(hookenv.charm_dir(), keyfile))
```

Comment:
Sure. The mock hookenv.charm_dir() only gets populated with symlinks to "metadata.yaml" and the "templates" directory - see FakeJuju's _setUp() method in charmtest/juju/fixture.py. In a real deployment, the entire charm's contents (including this key) would be present - so I manually added that additional symlink.